src/components/Cluster/NewCluster/CreateNodePoolsCluster.js

```diff
@@ -359,212 +220,160 @@ class CreateNodePoolsCluster extends Component {
-                            aria-hidden='true'
-                            onClick={() => this.removeNodePoolForm(npId)}
-                          />
-                        </AddNodePoolFlexColumnDiv>
-                      </AddNodePoolWrapperDiv>
-                    </SlideTransition>
-                  );
-                })}
-              </TransitionGroup>
-              <Button onClick={this.addNodePoolForm}>
-                <i className='fa fa-add-circle' /> ADD NODE POOL
-              </Button>
-              <hr style={{ margin: '30px 0' }} />
-            </WrapperDiv>
-
-            {this.state.error && CreateNodePoolsCluster.errorState()}
-
-            <FlexRowDiv>
-              <ErrorFallback error={this.props.clusterCreateError}>
-                <Button
-                  bsSize='large'
-                  bsStyle='primary'
-                  disabled={!this.isValid()}
-                  loading={submitting}
-                  onClick={this.createCluster}
-                  type='button'
-                >
-                  Create Cluster
-                </Button>
-              </ErrorFallback>
-              {/* We want to hide cancel button when the Create NP button has been clicked */}
-              {!submitting && (
-                <Button
-                  bsSize='large'
-                  bsStyle='default'
-                  loading={submitting}
-                  onClick={this.props.closeForm}
-                  type='button'
-                >
-                  Cancel
-                </Button>
+                  </InputGroup>
+                </div>
+              </MasterAZSelectionInput>
+            )}
+            <AZWrapperDiv>
+              {hasAZLabels && (
+                <AvailabilityZonesParser
+                  min={minAZ}
+                  max={maxAZ}
+                  defaultValue={defaultAZ}
+                  zones={this.props.availabilityZones}
+                  updateAZValuesInParent={this.updateAZ}
+                  isLabels={true}
+                  isRadioButtons
+                />
               )}
-            </FlexRowDiv>
-            <p style={{ maxWidth: '650px', margin: '0 auto' }}>
-              Note that it takes around 20 minutes on average until a new
-              cluster is fully available.
-            </p>
-          </>
-        </DocumentTitle>
-      </Breadcrumb>
+              {hasAZLabels && zonesArray.length < 1 && (
+                <p className='danger'>Please select one availability zone.</p>
+              )}
+              {hasAZLabels && zonesArray.length > maxAZ && (
+                <p className='danger'>
+                  {maxAZ} is the maximum you can have. Please uncheck{' '}
+                  {zonesArray.length - maxAZ} of them.
+                </p>
+              )}
+            </AZWrapperDiv>
+          </FlexColumn>
+          {Object.keys(nodePools).length === 0 && <HorizontalLine />}
+          <NodePoolsTransitionGroup>
```

Comment:
Do you think we can extract the Node Pools functionality to another component?

Reply:
Sure, but not in this PR ;)